### PR TITLE
(PUP-8463) Check modified Ruby files in PRs against 'ruby -wc'

### PR DESCRIPTION
**Pull request by jhelwig**

### CODE REVIEW

**Comment by jhelwig** [comment]
I'd like to get the remaining warnings cleaned up before this goes in, but I wanted to get this up so people could take a look, and start poking at it.

File: Rakefile [hunk 1]
```diff
@@ -110,6 +111,31 @@ task(:commits) do
...
+    puts "#{modified_file}"
+
+    stdout, stderr, _ = Open3.capture3("ruby -wc \"#{modified_file}\"")
+    unless stderr.empty?
+      ruby_files_ok = false
+      puts stderr
+    end
+    puts stdout
+  end
+  raise "One or more ruby files contain warnings." unless ruby_files_ok
```
Comment (joshcooper) [inline Rakefile]: Does `stderr` include the file name that generated the warning? If not, can you add it here?

File: Rakefile [hunk 2]
```diff
@@ -110,6 +111,31 @@ task(:commits) do
...
+  # accomplished by using the TRAVIS_COMMIT_RANGE environment variable, which
+  # is present in travis CI and populated with the range of commits the PR
+  # contains. If not available, this falls back to `master..HEAD` as a next
+  # best bet as `master` is unlikely to ever be absent.
+  commit_range = ENV['TRAVIS_COMMIT_RANGE'].nil? ? 'master...HEAD' : ENV['TRAVIS_COMMIT_RANGE']
+  ruby_files_ok = true
+  puts "Checking modified files #{commit_range}"
+  %x{git diff --name-only #{commit_range}}.each_line do |modified_file|
+    modified_file.chomp!
+    next unless File.extname(modified_file) == '.rb'
```
Comment (joshcooper) [inline Rakefile]: nit, I prefer `next if File.extname(...) != '.rb'` but maybe that's just me

File: Rakefile (hunk 1 above)
Comment (jhelwig) [reply Rakefile]: The output from the `puts stderr` up above will look something like the following:
```
spec/unit/pops/factory_spec.rb:194: warning: possibly useless use of == in void context
spec/unit/pops/factory_spec.rb:202: warning: possibly useless use of == in void context
spec/unit/pops/factory_spec.rb:214: warning: possibly useless use of == in void context
spec/unit/pops/factory_spec.rb:224: warning: possibly useless use of == in void context
spec/unit/pops/factory_spec.rb:232: warning: possibly useless use of == in void context
```
So, we'll actually get final output that looks something like:
```
spec/unit/pops/factory_spec.rb
spec/unit/pops/factory_spec.rb:194: warning: possibly useless use of == in void context
spec/unit/pops/factory_spec.rb:202: warning: possibly useless use of == in void context
spec/unit/pops/factory_spec.rb:214: warning: possibly useless use of == in void context
spec/unit/pops/factory_spec.rb:224: warning: possibly useless use of == in void context
spec/unit/pops/factory_spec.rb:232: warning: possibly useless use of == in void context
Syntax OK
```

The first line being from `puts "#{modified_file}"`, the last from `puts stdout`, and everything in the middle from `puts stderr`.

File: Rakefile (hunk 2 above)
Comment (jhelwig) [reply Rakefile]: I don't have a strong opinion either way in this particular case. If the condition were any more complex than it is here, I'd definitely prefer the `next if ...` form, though.

File: Rakefile (hunk 1 above)
Comment (jhelwig) [reply Rakefile]: Though, looking at this, there's absolutely no reason to do the string interpolation for `puts "#{modified_file}"`. Bash-isms must have been creeping into my brain there.

**Pull request merged**